### PR TITLE
Implement Action State & Integrate All Battle States

### PR DESCRIPTION
**Pull request by krittintrs**
- Refactor wrong spelling attributes & method
- Refactor redundant functions (mostly in Entity, Card, Buff, and Effect class)
- Change logic of existing state to match the state diagram
- Implement logic of action, resolve, effect selection, and end stats

State Diagram: https://www.figma.com/board/FbcG7K9YSG8BpABLqeEFgo/Re%3ASacrifice-State-Diagram?node-id=0-1&t=nonQkC20ZbVtAzN6-1

### CODE REVIEW

File: src/cardSystem/Entity.py [hunk 1]
```diff
@@ -2,29 +2,85 @@
...
+
+        # reset stats
+        self.attack = 0
+        self.defense = 0
+        self.speed = 0
+        self.range = 0
+
+        # reset buffs
+        for buff in self.buffs:
+            buff.turn_passed()
```
Comment (phukaokub) [inline src/cardSystem/Entity.py:73]: I don't think `reset` method would be the naming of the passing turn. Maybe we used another words to indicate it to reduce confusion?

Comment (krittintrs) [reply src/cardSystem/Entity.py:73]: agree, I will change to `turn_passed`

Comment (krittintrs) [reply src/cardSystem/Entity.py:73]: For clarity, I think i will change to `next_turn` instead.

**Review by phukaokub — APPROVED** [review]
After merging this PR, we have to create another issued on the Buff logic to execute before selection state to render the BUFF, DEBUFF numbers on card.

**Review by rymTooo — APPROVED** [review]
Look good to me TT

**Pull request merged**